app/backend/src/clj/teet/integration/x_road.clj

```diff
@@ -155,16 +155,73 @@
               :omandiosa_suurus (omandi-get ox :a:omandiosa_suurus)})
            owner-xml-seq)}))
 
+(defn d-jagu34* [key children sequential-vals?]
+  (let [leaf? (fn [m] (->> m
+                           :content
+                           (every? (complement map?))))
+        leaves (filterv leaf? children)
+        subs (filterv (complement leaf?) children)
+        tag-without-prefix #(some->> %
+                             :tag
+                             name
+                             (partition-by (partial = \:))
+                             last
+                             clojure.string/join
+                             keyword)
+        joined-content (comp (partial clojure.string/join "\n") :content)
+        recursed-map (apply merge
+                            (mapv (fn [s]
+                                    (d-jagu34* (tag-without-prefix s) (:content s) true)) subs))
+        leaves-map (into {} (mapv
+                      (juxt tag-without-prefix joined-content)
+                      leaves))
+        maybe-vectorise (if sequential-vals?
+                          (fn [val] [val])
+                          identity)]
+    
+    {key (maybe-vectorise
+          (merge recursed-map
+                 leaves-map))}))
 
 (defn d-jagu34 [k-xml p1 p2]
-  (let [j3-seq (z/xml-> k-xml p1 p2)]
-    {p1
-     (into [] (for [j3 j3-seq]
-                {:kande_alguskuupaev (z/xml1-> j3 :a:kande_alguskuupaev z/text)
-                 :kande_liik (z/xml1-> j3 :a:kande_liik z/text)
-                 :kande_liik_tekst (z/xml1-> j3 :a:kande_liik_tekst z/text)
-                 :registriosa_nr (z/xml1-> j3 :a:registriosa_nr z/text)
-                 :oiguse_liik_tekst (z/xml1-> j3 :a:oiguse_liik_tekst z/text)}))}))
+  (let [;; children will have eg subtree Jagu3  elements (from under the containing jagu3 element)
+        children (z/xml-> k-xml p1 clojure.zip/children)
+        jagu-maps (mapv #(d-jagu34* p1 (:content %) false)
+                        children)]
+    {p1 (mapv p1 jagu-maps)}))
+
+
+
+(def jagu34-summary-fields [:kande_liik_tekst
+                            :kande_kehtivus
+                            :kande_tekst
+                            :oiguse_seisund_tekst])
+
+(defn summary-jagu34-only [k-resp]
+  ;; keeps only jagu34-summary-fields in the maps the maps in the seq under :jagu3 and :jagu4.
+  (let [map-fn (fn [ms]
+                 (map #(select-keys % jagu34-summary-fields)
+                      ms))]
+    (-> k-resp
+        (update :jagu3 map-fn)
+        (update :jagu4 map-fn))))
+
+(defn active-jagu34-only [k-resp]
+  ;; prunes "lopetatud" records from the maps in the seq under :jagu3 and :jagu4.
+  ;; {:jagu3 [... {:oiguse_seisund "L", ..} ...]
+  (let [filter-fn #(filter (comp (partial not= "L") :oiguse_seisund) %)]
+    (-> k-resp
+        (update :jagu3 filter-fn)
+        (update :jagu4 filter-fn))))
+
+(defn parse-kande-tekst [kande-tekst-str]
```

Comment:
If we only need to show the HTML content, I would try just not parsing it and showing it as is with `dangerouslySetInnerHTML` I don't think any scripts are coming from property register.